File: MyPersonalIndex/Classes/Queries/AAQueries.cs
using System;
using System.Data;
using System.Data.SqlServerCe;

namespace MyPersonalIndex
{
    class AAQueries : Queries
    {
        public static QueryInfo DeleteAA(int Portfolio, string AAin)
        {
            return new QueryInfo(
                string.IsNullOrEmpty(AAin) ?
                   "DELETE FROM AA WHERE Portfolio = @Portfolio" :
                    string.Format("DELETE FROM AA WHERE Portfolio = @Portfolio AND ID NOT IN ({0})", AAin),
                new SqlCeParameter[] { 
                    AddParam("@Portfolio", SqlDbType.Int, Portfolio)
                }
            );
        }

        public static QueryInfo InsertAA(int Portfolio, string AA, double? Target)
        {
            return new QueryInfo(
                "INSERT INTO AA (Portfolio, AA, Target) VALUES (@Portfolio, @AA, @Target)",
                new SqlCeParameter[] {
                    AddParam("@Portfolio", SqlDbType.Int, Portfolio),
                    AddParam("@AA", SqlDbType.NVarChar, AA),
                    AddParam("@Target", SqlDbType.Decimal, Target.HasValue ? Target : (object)System.DBNull.Value)
                }
            );
        }

        public static QueryInfo UpdateAA(int ID, string AA, double? Target)
        {
            return new QueryInfo(
                "UPDATE AA SET AA = @AA, Target = @Target WHERE ID = @ID",
                new SqlCeParameter[] {
                    AddParam("@ID", SqlDbType.Int, ID),
                    AddParam("@AA", SqlDbType.NVarChar, AA),
                    AddParam("@Target", SqlDbType.Decimal, Target.HasValue ? Target : (object)System.DBNull.Value)
                }
            );
        }
    }
}

File: MyPersonalIndex/classes/queries/aaQueries.cpp
#include "aaQueries.h"
#include "queries.h"
#include "globals.h"

queries::queryInfo* aaQueries::updateAA(globals::assetAllocation* const aa)
{
    if (!aa)
        return 0;

    QVariant target(QVariant::Double);
    if (target != -1)
        target = aa->target;

    if(aa->id == -1) // insert new
    {
        return new queryInfo(
            "INSERT INTO AA (PortfolioID, Description, Target)"
            " VALUES (:PortfolioID, :Description, :Target)",
            QList<parameter>()
                << parameter(":PortfolioID", m_portfolioID)
                << parameter(":Description", aa->description)
                << parameter(":Target", target)
        );
    }
    else // update
    {
        return new queryInfo(
            "UPDATE AA SET Description = :Description, Target = :Target WHERE ID = :AAID",
            QList<parameter>()
                << parameter(":Description", aa->description)
                << parameter(":Target", target)
                << parameter(":AAID", aa->id)
        );
    }
}

queries::queryInfo* aaQueries::deleteAA(const int &aaID)
{
    return new queryInfo(
            "DELETE FROM AA WHERE ID = :AAID",
            QList<parameter>()
                << parameter(":AAID", aaID)
        );
}
